pkg/cmd/variable/create/create.go

```diff
@@ -0,0 +1,219 @@
+	"github.com/spf13/cobra"
+)
+
+func NewCmdCreate(f *cmdutil.Factory, runF func(*shared.PostPatchOptions) error) *cobra.Command {
+	opts := &shared.PostPatchOptions{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		HttpClient: f.HttpClient,
+		Prompter:   f.Prompter,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create <variable-name>",
+		Short: "Create variables",
+		Long: heredoc.Doc(`
+			Create a value for a variable on one of the following levels:
+			- repository (default): available to Actions runs in a repository
+			- environment: available to Actions runs for a deployment environment in a repository
+			- organization: available to Actions runs within an organization
+
+			Organization variable can optionally be restricted to only be available to
+			specific repositories.
+		`),
+		Example: heredoc.Doc(`
+			# Add variable value for the current repository in an interactive prompt
+			$ gh variable create MYVARIABLE
+
+			# Read variable value from an environment variable
+			$ gh variable create MYVARIABLE --body "$ENV_VALUE"
+
+			# Read variable value from a file
+			$ gh variable create MYVARIABLE < myfile.txt
+
+			# Create variable for a deployment environment in the current repository
+			$ gh variable create MYVARIABLE --env myenvironment
+
+			# Create organization-level variable visible to both public and private repositories
+			$ gh variable create MYVARIABLE --org myOrg --visibility all
+
+			# Create organization-level variable visible to specific repositories
+			$ gh variable create MYVARIABLE --org myOrg --repos repo1,repo2,repo3
+
+		`),
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if err := cmdutil.MutuallyExclusive("specify only one of `--org` or `--env`", opts.OrgName != "", opts.EnvName != ""); err != nil {
+				return err
+			}
+
+			if err := cmdutil.MutuallyExclusive("specify only one of `--body`", opts.Body != ""); err != nil {
+				return err
+			}
+
+			opts.VariableName = args[0]
+
+			if cmd.Flags().Changed("visibility") {
+				if opts.OrgName == "" {
+					return cmdutil.FlagErrorf("`--visibility` is only supported with `--org`")
+				}
+
+				if opts.Visibility != shared.Selected && len(opts.RepositoryNames) > 0 {
+					return cmdutil.FlagErrorf("`--repos` is only supported with `--visibility=selected`")
+				}
+
+				if opts.Visibility == shared.Selected && len(opts.RepositoryNames) == 0 {
+					return cmdutil.FlagErrorf("`--repos` list required with `--visibility=selected`")
+				}
+			} else {
+				if len(opts.RepositoryNames) > 0 {
+					opts.Visibility = shared.Selected
+				}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return createRun(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.OrgName, "org", "o", "", "Create `organization` variable")
+	cmd.Flags().StringVarP(&opts.EnvName, "env", "e", "", "Create deployment `environment` variable")
```

Comment:
NIT - any reason why we're calling it deployment environment variable?

Reply:
Tokens are usually created based on deployment configs is my guess. But I am not sure.